applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -50,41 +64,61 @@ def __init__(self, identifier, response_settings, model_part = None):
         self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(ProjectParametersPrimal, model_part)
         self.primal_analysis.GetModelPart().AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
 
+    def RunCalculationProcess(self, calculate_gradient):
```

Comment:
This function could actually be implemented by the base class, because it will look the same for all derived ones.